docs/sdk2/usage/STATE_SYNC.md

```diff
@@ -0,0 +1,417 @@
+ </properties>
+
+ <dependencies>
+     <dependency>
+       <groupId>com.commercetools.sdk</groupId>
+       <artifactId>commercetools-http-client</artifactId>
+       <version>${commercetools.version}</version>
+     </dependency>
+     <dependency>
+       <groupId>com.commercetools.sdk</groupId>
+       <artifactId>commercetools-sdk-java-api</artifactId>
+       <version>${commercetools.version}</version>
+     </dependency>
+ </dependencies>
+
+```
+
+### Client configuration and creation
+
+For client creation use [ClientConfigurationUtils](#todo) which apply the best practices for `ProjectApiRoot` creation.
+If you have custom requirements for the client creation make sure to replace `SphereClientFactory` with `ApiRootBuilder` as described in this [Migration Document](https://docs.commercetools.com/sdk/java-sdk-migrate#client-configuration-and-creation).
+
+### Signature of StateSyncOptions
+
+As models and update actions have changed in the JVM-SDK-V2 the signature of SyncOptions is different. It's constructor now takes a `ProjectApiRoot` as first argument. The callback functions are signed with `State`, `StateDraft` and `StateUpdateAction` from `package com.commercetools.api.models.state.*`
+
+> Note: Type `StateUpdateAction` has changed to `StateUpdateAction`. Make sure you create and supply a specific StateUpdateAction in `beforeUpdateCallback`. For that you can use the [library-utilities](#todo) or use a JVM-SDK builder ([see also](https://docs.commercetools.com/sdk/java-sdk-migrate#update-resources)):
+
+```java
+// Example: Create a state update action to change name taking the 'newName' of the stateDraft
+    final Function<LocalizedString, StateUpdateAction> createBeforeUpdateAction =
+        (newName) -> StateSetNameAction.builder().name(newName).build();
+
+// Add the change name action to the list of update actions before update is executed
+    final TriFunction<
+            List<StateUpdateAction>, StateDraft, State, List<StateUpdateAction>>
+        beforeUpdateStateCallback =
+            (updateActions, newStateDraft, oldState) -> {
+              final StateUpdateAction beforeUpdateAction =
+                  createBeforeUpdateAction.apply(newStateDraft.getName());
+              updateActions.add(beforeUpdateAction);
+              return updateActions;
+            };
+```
+
+### Build StateDraft (syncing from external project)
+
+The state-sync expects a list of `StateDraft`s to process. If you use java-sync-library to sync your states from any external system into a commercetools platform project you have to convert your data into CTP compatible `StateDraft` type. This was done in previous version using `DraftBuilder`s.
+The V2 SDK do not have inheritance for `DraftBuilder` classes but the differences are minor and you can replace it easily. Here's an example:
+
+```java
+// StateDraftBuilder in v1 takes parameters 'name', 'cartPredicate', 'value', 'target', 'sortOrder', 'requiresDiscountCode' 
```

Comment:
Adjust the comment

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1044/commits/a3b6146ca828af2079f8991e138b8ce5124dce3e